specification/containerservice/Fleet.Management/fleetnamespace.tsp

```diff
@@ -120,28 +121,241 @@ union EgressPolicy {
   string,
 }
 
-@doc("The profile of the fleet managed namespace.")
+@doc("The possible values representing different network policy rules.")
+union PolicyRule {
+  @doc("Allow all network traffic.")
+  AllowAll: "AllowAll",
+
+  @doc("Deny all network traffic.")
+  DenyAll: "DenyAll",
+
+  @doc("Allow traffic within the same namespace.")
+  AllowSameNamespace: "AllowSameNamespace",
+
+  string,
+}
+
 @added(Versions.v2025_03_02_naar)
-model FleetManagedNamespaceProfile {
-  @doc("Action if Kubernetes managed namespace with the same name already exists.")
-  adoptionPolicy?: AdoptionPolicy;
+@doc("The orchestrator profile to manage the lifecycle of the namespace.")
+model OrchestratorProfile {
+  @doc("The type of the policy to be used.")
+  type: OrchestratorProfileType;
 
-  @doc("Delete options of a fleet managed namespace.")
-  deletePolicy?: DeletePolicy;
+  @doc("The policy configuration for the type of the orchestrator.")
+  policy: OrchestratorPolicy;
+}
 
-  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "Records is the most semantically correct choice for this use-case"
-  @doc("The labels for the fleet managed namespace.")
-  labels?: Record<string>;
+@added(Versions.v2025_03_02_naar)
+@doc("The type of the namespace orchestrator.")
+union OrchestratorProfileType {
+  @doc("Using a fleet hub and kubefleet ClusterResourcePlacement.")
+  Hub: "Hub",
+
+  string,
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("The available orchestrator type.")
+union OrchestratorPolicy {
+  @doc("The Hub Orchestrator.")
+  hubPolicy: HubPolicy,
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("HubPolicy uses a fleet hub and CRP to propagate the namespace.")
+model HubPolicy {
+  @doc("The default ClusterResourcePlacement (CRP) policy configuration.")
+  defaultClusterResourcePlacement: ClusterResourcePlacement;
+}
+
+@added(Versions.v2025_03_02_naar)
+@doc("The desired state of ClusterResourcePlacement")
+model ClusterResourcePlacement {
+  @doc("Policy defines how to select member clusters to place the selected resources. If unspecified, all the joined member clusters are selected.")
+  policy?: ClusterResourcePlacementPolicy;
+}
```

Comment:
What is the reason to have so many levels of indirections?

hubpolicy contains one ClusterResourcePlacement which contains one ClusterResourcePlacementPolicy.

Will this work?
```
PropagationPolicy {
  type: Placement | Direct
  PlacementProfile | DirectProfile
}

// This is the same as ClusterResourcePlacementSpec.
PlacementProfile {
  PlacementPolicy // == PlacementPolicy in ClusterResourcePlacementSpec.
}
```

Reply:
The reason for the nesting here is:
a) match the naming/approach for **default**ResourceQuota, **default**NetworkPolicy
b) allow an extension point if in the future we want the placement propagation policy to place something else in addition to or instead of a CRP.

My personal take is that those reasons outweigh the nesting annoyance; WDYT @circy9 ?